code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -113,6 +113,93 @@
 		return
 	return held_item.GetID()
 
+/**
+ * Used to fetch all the cash a vendor can access from the human
+ * See /mob/living/carbon/human/proc/get_cash_list() proc for the list of items and inventory slots it searches through
+ */
+/mob/living/carbon/human/proc/get_accessible_cash(list/cash_list = null)
+	var/available_cash = 0
+	if(!cash_list)
+		cash_list = get_cash_list()
+	if(!length(cash_list))
+		return 0
+	for(var/found_item in cash_list)
+		if(istype(found_item, /obj/item/holochip))
+			var/obj/item/holochip/chip_stack = found_item
+			available_cash += chip_stack.credits
+		if(istype(found_item, /obj/item/card/id))
+			var/obj/item/card/id/id_card = found_item
+			available_cash += id_card.registered_account.account_balance
+	return available_cash
+
+/**
+ * Used by vendors to see if the human can spend a certain amount of cash.
+ * Returns FALSE if they cannot, TRUE if they can and withdraws the cash.
+ * Arguments:
+ * * to_spend = how much cash needs to be deducted
+ */
+/mob/living/carbon/human/proc/spend_cash(to_spend)
+	if(!to_spend)
+		return FALSE
+	var/list/cash_list = get_cash_list()
+	if(!length(cash_list)) //We have no accessible cash items, early return
+		return FALSE
+	if(to_spend > get_accessible_cash(cash_list)) //If we don't have enough money, early return
+		return FALSE
```

Comment:
put a comment before for loop
`// these 'for loop' are intentionally separated into two sections because we're going to detect holochips first, then detect cards later.`